### DISCUSSION TITLE
Uno platform supported Brushes

### DISCUSSION THREAD

**Discussion by limeniye** [unoplatform/uno#9840] [General]
I'm making theme changer package:
https://github.com/ligric/lungo

Right now I'm looking for a way to add a theme change effect for Uno plafrom.
For WPF I did it through VisualBrush, But for UWP I should to use Compositor.

I wrote a simple example with Compositor (without Win2D and DirectX).
But my example doesn't works on all platforms.

![зображення](https://user-images.githubusercontent.com/69314237/190190345-b3f3db6b-14be-4387-8de2-4e9f9d894076.png)

- [x] Uno,Windows
- [x] Uno.Skia.Wpf
- [ ] Uno,Wasm
- [ ] Uno,Mobile

Are there any alternatives to implement such an effect in Wasm, for example?
Are there plans to support Compositor brushes for other platforms?

**Comment by jeromelaban** [comment]
Thanks for the report. 

At this time, Composition is mostly supported on Skia (the rendering engine uses it for the UI). The other platforms (iOS, Android, Wasm) don't have support for it at this time. 

Another way to have rendering similar to Composition is to use [SkiaSharp](https://github.com/mono/SkiaSharp), which is supported on all platforms.

**Reply by limeniye** [reply]
@jeromelaban, I made SKBitmap.
I searched for Brushes, but if I understand correctly, there is no such feature in SkiaSharp.
So I should to display my result on SKCanvasView.

But I see it only for Desktop, Gtk, WPF
![зображення](https://user-images.githubusercontent.com/69314237/190495590-4b4fd968-d9d6-4d78-9227-a78b3ba118bf.png)

Ok I'm trying to use it for Desktop but there is empty
![зображення](https://user-images.githubusercontent.com/69314237/190495797-623fcf3e-db53-4edb-8809-e2932694e85f.png)

**Reply by jeromelaban** [reply]
To use SkiaSharp, see a small example here: https://github.com/unoplatform/Uno.Samples/tree/master/UI/SkiaSharpTest

**Reply by limeniye** [reply]
Thank you for Test solution.
But its only UWP example in Uno.Xamarin solution. 
And it doesn't work with Wasm.
![зображення](https://user-images.githubusercontent.com/69314237/190499470-15ff378b-bb9b-4972-a5b7-3bea8ef5488e.png)

I have Uno.WinUI solution and skia even doesn't work with Windows platform:
```XML
xmlns:skia="using:SkiaSharp.Views.Desktop"
```

```XML
<skia:SKXamlCanvas>
            
</skia:SKXamlCanvas>
```

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	XLS0414	The type 'skia:SKXamlCanvas' was not found. Verify that you are not missing an assembly reference and that all referenced assemblies have been built.	ForDeleteUno.WinUI.Windows	C:\Users\limeniye\Desktop\ForDeleteUno.WinUI\ForDeleteUno.WinUI\ForDeleteUno.WinUI.Shared\MainPage.xaml	26	

```

My packages
```XML
<PackageReference Include="SkiaSharp" Version="2.88.2" />
<PackageReference Include="SkiaSharp.Views" Version="2.88.2" />
<PackageReference Include="SkiaSharp.Views.Uno.WinUI" Version="2.88.2" />
```

**Reply by jeromelaban** [reply]
Here's another sample which uses SkiaSharp with WinUI: https://github.com/unoplatform/Uno.Samples/tree/master/UI/SkottieSample/SkottieSample
